test/export/StructureDefinitionExporter.test.ts

```diff
@@ -496,6 +496,129 @@ describe('StructureDefinitionExporter', () => {
     );
   });
 
+  it('should apply a correct OnlyRule on a non-reference FSHy choice', () => {
+    const profile = new Profile('Foo');
+    profile.parent = 'Observation';
+
+    const targetProfile = new Profile('MySpecialQuantity');
+    targetProfile.parent = 'Quantity';
+    doc.profiles.set(targetProfile.name, targetProfile);
+
+    const rule = new OnlyRule('value[x]');
+    rule.types = [{ type: 'MySpecialQuantity' }];
+    profile.rules.push(rule);
+
+    exporter.exportStructDef(profile);
+    const sd = exporter.structDefs[1];
+    const baseStructDef = resolve('Observation');
+
+    const baseValue = baseStructDef.findElement('Observation.value[x]');
+    const constrainedValue = sd.findElement('Observation.value[x]');
+
+    expect(baseValue.type).toHaveLength(11);
+    expect(baseValue.type[0]).toEqual({ code: 'Quantity' });
+    expect(baseValue.type[1]).toEqual({ code: 'CodeableConcept' });
+    expect(baseValue.type[2]).toEqual({ code: 'string' });
```

Comment:
These three need to be updated now that @jafeltra's PR with the new `ElementDefinitionType` class is merged.  For example, line 519 should be
```ts
expect(baseValue.type[0]).toEqual(new ElementDefinitionType('Quantity'));
```

Reply:
I won't specifically call out the others, but anywhere in these new tests where you're comparing against the `type`, you'll need to use an instance of the `ElementDefinitionType` class instead.